cli/src/lib.rs

```diff
@@ -497,6 +432,71 @@ impl Iroha {
     }
 }
 
+/// Combine configs from as many sources as are available.
+///
+/// # Errors
+/// If a necessary field isn't specified or specified with an error in
+/// the overriding structure, report an error
+pub fn combine_configs(args: &Arguments) -> Result<Configuration, color_eyre::Report> {
+    let config = {
+        let colors = style::Styling::new();
+        let config = args
+            .config_path
+            .first_existing_path()
+            .map_or_else(
+                || {
+                    eprintln!(
+                        "Configuration file not found at {}.\n{}\n",
+                        args.config_path.style(colors.highlight),
+                        "Falling back to environment variables".style(colors.highlight),
+                    );
+                    ConfigurationProxy::default()
+                },
+                |path| ConfigurationProxy::from_path(&path.as_path()),
+            )
+            .override_with(ConfigurationProxy::from_env())
+            .build();
+
+        match config {
+            Ok(mut config) => {
+                if style::should_disable_color() {
+                    config.disable_panic_terminal_colors = true;
+                    // Remove terminal colors to comply with XDG
+                    // specifications, Rust's conventions as well as remove
+                    // escape codes from logs redirected from STDOUT. If you
+                    // need syntax highlighting, use JSON logging instead.
+                    config.logger.terminal_colors = false;
+                }
+                config
+            }
+            Err(e) => {
+                use iroha_config::base::derive::Error::*;
+
+                color_eyre::install()?;
```

Comment:
Do we really need to call it here as long as the hook is already installed in `cli/src/main.rs` and that's its only call site?

Reply:
We do. Courtesy of `eyre` installing a default hook when a `Report` is constructed, the hook installation being irreversible and us having the brilliant foresight of introducing a config parameter for colors, rather than using XDG standards, we actually have to initialise the correct hook **each** time we return an error.